stdlib/statistics.pyi

```diff
@@ -98,15 +101,15 @@ class NormalDist:
     if sys.version_info >= (3, 9):
         def zscore(self, x: float) -> float: ...
 
-    def __eq__(self, x2: object) -> bool: ...
-    def __add__(self, x2: float | NormalDist) -> NormalDist: ...
-    def __sub__(self, x2: float | NormalDist) -> NormalDist: ...
-    def __mul__(self, x2: float) -> NormalDist: ...
-    def __truediv__(self, x2: float) -> NormalDist: ...
-    def __pos__(self) -> NormalDist: ...
-    def __neg__(self) -> NormalDist: ...
+    def __eq__(x1, x2: NormalDist) -> bool: ...  # type: ignore[override]
```

Comment:
the change to the `x2` parameter type here is incorrect. The change you're making here would lead to type checkers inferring "it will literally raise an exception if you do an equality comparison between a `NormalDist` instance and an instance of something that isn't a `NormalDist`". But that's not true: the [implementation](https://github.com/python/cpython/blob/39b2f82717a69dde7212bc39b673b0f55c99e6a3/Lib/statistics.py#L1093-L1097) is careful to `return NotImplemented` rather than raising an exception if `x2` isn't an instance of `NormalDist`. That means that the `==` comparison will gracefully evaluate to `False` if `x2` isn't an instance of `NormalDist`.

```suggestion
    def __eq__(x1, x2: object) -> bool: ...
```

Reply:
Ahh of course, I was thinking of the `NotImplemented` behavior in the binops like `__add__` 👌🏻